chain/proofs/proofs.go

```diff
@@ -19,48 +19,24 @@ func GenerateUnsealedCID(proofType abi.RegisteredSealProof, pieces []abi.PieceIn
 	if err != nil {
 		return cid.Undef, err
 	}
+	upssize := abi.PaddedPieceSize(ssize).Unpadded()
 
-	pssize := abi.PaddedPieceSize(ssize)
-	allPieces := make([]abi.PieceInfo, 0, len(pieces))
 	if len(pieces) == 0 {
-		allPieces = append(allPieces, abi.PieceInfo{
-			Size:     pssize,
-			PieceCID: zerocomm.ZeroPieceCommitment(pssize.Unpadded()),
-		})
-	} else {
-		var sum abi.PaddedPieceSize
-
-		padTo := func(pads []abi.PaddedPieceSize) {
-			for _, p := range pads {
-				allPieces = append(allPieces, abi.PieceInfo{
-					Size:     p,
-					PieceCID: zerocomm.ZeroPieceCommitment(p.Unpadded()),
-				})
-
-				sum += p
-			}
-		}
-
-		for _, p := range pieces {
-			ps, _ := getRequiredPadding(sum, p.Size)
-			padTo(ps)
-
-			allPieces = append(allPieces, p)
-			sum += p.Size
-		}
+		return zerocomm.ZeroPieceCommitment(upssize), nil
+	}
 
-		ps, _ := getRequiredPadding(sum, pssize)
-		padTo(ps)
+	pcid, psz, err := commp.PieceAggregateCommP(proofType, pieces)
+	if err != nil {
+		return cid.Undef, err
 	}
 
-	// Next we need to generate the unsealed CID by merkleizing the pieces; even though this function
-	// has the same name, it does a different job and won't give us the desired results if we hadn't
-	// done the piece-padding first.
-	return nonffi.GenerateUnsealedCID(proofType, allPieces)
+	return commp.ZeroPadPieceCommitment(pcid, psz.Unpadded(), upssize)
```

Comment:
ok, so this works because even though `PieceAggregateCommP` doesn't pad out a sector, `ZeroPadPieceCommitment` will defer to go-fil-comp-hashhash's `PadCommP` do to its padding using the static `stackedNulPadding` to fill out the missing portions of the tree

it's quite a departure from the approach here, but appears to work fine and has been tested in production elsewhere.

My only quibble is the "experimental" notice in go-fil-commp-hashhash which I'd love for you to remove - you're an admin of the package and can make that call.